### PR TITLE
CLI: auth

### PR DESCRIPTION
**Pull request by gimenete**
```
./bin/run.js auth login
./bin/run.js auth logout
./bin/run.js auth status
```

TODO: tests

### CODE REVIEW

File: cli/src/commands/auth/login.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import prompts from 'prompts';
+import { BaseCommand } from '../../base.js';
+import { readAPIKeyFromFile, writeAPIKey } from '../../key.js';
+
+export default class Login extends BaseCommand {
+  static description = 'Authenticate with Xata';
+
+  static examples = [];
+
+  static flags = {};
```
Comment (SferaDev) [inline cli/src/commands/auth/login.ts:10]: Add a flag to login without interactivity

Comment (gimenete) [reply cli/src/commands/auth/login.ts:10]: That'd be handy, but my concern is that the API key will be stored in the command history in the shell.

File: cli/src/commands/auth/login.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  async run(): Promise<void> {
+    const existingKey = await readAPIKeyFromFile();
+    if (existingKey) {
+      const result = await prompts({
+        type: 'confirm',
+        name: 'overwrite',
+        message: 'Authentication is already configured, do you want to overwrite it?'
+      });
+      if (!result.overwrite) this.exit(2);
```
Comment (SferaDev) [inline cli/src/commands/auth/login.ts:22]: Are we documenting error codes somewhere?

**Review by SferaDev — APPROVED** [review]

File: cli/src/commands/auth/login.ts (hunk 2 above)
Comment (gimenete) [reply cli/src/commands/auth/login.ts:22]: Good point. I'll add a note on the issue for improvements.

**Pull request merged**